### PR TITLE
Don't warn when boxing large arrays r? @mcarton

### PR DESCRIPTION
**Pull request by Manishearth**

### CODE REVIEW

**Comment by mcarton** [comment]
What about `[[ComplexF64; 50]; 50]`?

**Comment by Manishearth** [comment]
I guess we should use the ty::layout module eddyb was talking about

**Comment by Manishearth** [comment]
Updated.

File: clippy_lints/src/escape.rs [hunk 1]
```diff
@@ -170,3 +179,21 @@ impl<'a, 'tcx: 'a> Delegate<'tcx> for EscapeDelegate<'a, 'tcx> {
...
+
+impl<'a, 'tcx: 'a+'gcx, 'gcx: 'a> EscapeDelegate<'a, 'tcx, 'gcx> {
+    fn is_large_box(&self, ty: ty::Ty<'gcx>) -> bool {
+        // Large types need to be boxed to avoid stack
+        // overflows.
+        match ty.sty {
+            ty::TyBox(ref inner) => {
+                if let Ok(layout) = inner.layout(self.infcx) {
+                    let size = layout.size(&self.target);
+                    size.bytes() > 1000
```
Comment (eddyb) [inline clippy_lints/src/escape.rs]: I would make this configurable or at least put it in a named constant.

Comment (Manishearth) [inline clippy_lints/src/escape.rs]: done

**Comment by Manishearth** [comment]
Passes build again, too. 😄

File: clippy_lints/src/escape.rs [hunk 2]
```diff
@@ -1,18 +1,23 @@
...
 use rustc::util::nodemap::NodeSet;
 use syntax::ast::NodeId;
 use syntax::codemap::Span;
 use utils::span_lint;
 
 pub struct Pass;
 
+/// Largest boxed type we warn about
+/// Larger types should not be on the stack anyway
+const TOO_LARGE_FOR_STACK: u64 = 1000;
```
Comment (mcarton) [inline clippy_lints/src/escape.rs]: This could have been in `clippy.toml` :smile:

Comment (Manishearth) [inline clippy_lints/src/escape.rs]: done

**Pull request merged**